lib/project/result-view.js

```diff
@@ -111,47 +111,33 @@ class ResultView {
 
   renderList() {
     const children = [];
-    let itemTopPosition = this.pathDetailsHeight;
-    let i = 0;
-
-    for (; i < this.matches.length; i++) {
-      let itemBottomPosition = itemTopPosition + this.matchHeight;
-      if (itemBottomPosition > this.top) break;
-      itemTopPosition = itemBottomPosition;
-    }
-
-    for (; i < this.matches.length; i++) {
+    const top = Math.max(0, this.top - this.pathDetailsHeight);
+    let itemTopPosition = 0;
+    for (let i = Math.floor(top / this.matchHeight); i < this.matches.length; i++) {
+      itemTopPosition = i * this.matchHeight;
       const match = this.matches[i];
       children.push(
-        etch.dom.div(
-          {
-            style: {
-              position: 'absolute',
-              height: this.matchHeight + 'px',
-              minWidth: '100%',
-              top: itemTopPosition - this.pathDetailsHeight + 'px'
-            },
-            key: i
-          },
-          etch.dom(MatchView, {
-            match,
-            regex: this.regex,
-            replacePattern: this.replacePattern,
-            isSelected: (i === this.selectedMatchIndex),
-            previewStyle: this.previewStyle
-          })
-        )
+        etch.dom(MatchView, {
+          match,
+          key: i,
+          regex: this.regex,
+          replacePattern: this.replacePattern,
+          isSelected: (i === this.selectedMatchIndex),
+          previewStyle: this.previewStyle,
+          top: itemTopPosition
+        })
```

Comment:
Curious about your reasoning for this. It seems like now, components like `ResultView` and `MatchView` are much more coupled to the fact that we use them inside of a `ListView`, and the `ListView` has a less generic interface. Was the `div` causing a problem for styling or something?